src/plugins/lightscript.js

```diff
@@ -121,7 +121,16 @@ pp.parseArrayComprehension = function (node) {
   const loop = this.startNode();
   node.loop = this.parseForStatement(loop);
   this.expect(tt.bracketR);
-  return this.finishNode(node, "ArrayComprehension");
+  node.comprehends = "array";
+  return this.finishNode(node, "Comprehension");
```

Comment:
I think I'd rather stick with two node types, `ArrayComprehension` and `ObjectComprehension`. Sound ok to you?